tools/shared.py

```diff
@@ -799,6 +799,24 @@ def do_replace(input_, pattern, replacement):
   return input_.replace(pattern, replacement)
 
 
+def read_file(file_path):
+  """Read from a file opened in text mode"""
+  with open(file_path) as fh:
+    return fh.read()
+
+
+def read_binary(file_path):
+  """Read from a file opened in binary mode"""
+  with open(file_path, 'rb') as fh:
+    return fh.read()
+
+
+def write_file(file_path, text):
+  """Write to a file opened in text mode"""
+  with open(file_path, 'w') as fh:
+    fh.write(text)
```

Comment:
Any particular reason you chose to use context managers here rather than `Path`?   I guess its slightly faster because it doesn't have to do the normalization stuff?

Reply:
Well no particular reason not to use `open` either. It's still the canonical way to open files. `Path` is shorter indeed, but `open` would be easier to adapt to specify the `size` parameter, for instance in `read`, which is used in a few paces of the code base.